### ISSUE TITLE
Batch convert for SUP files (i.e. OCR'ing)

### ISSUE THREAD

**Issue by adem0x** [SubtitleEdit/subtitleedit#3372]
SubtitleEdit has been great for dealing with subtitles.

The one thing I am missing is a way to do OCR'in SUP files. I chekced but, couldn't find this particular functionality --if it does exist, please tell me how I use it and then ignore this question :)

Anyway.

Here is what I mean:

Suppose I have a set of episodes for the same series.

A lot of the unusual words as well as OCR errors are common in these files.

So, if I could have a way to do all of these in a batch, I wouldn't have to do the same corrections over and over again.

Thank you again for this great utility.

**Comment by niksedk** [comment]
@adem0x: I think this works both via command line or Tools -> Batch convert...

**Comment by adem0x** [comment]
I am sorry for replying so late but I just had the opportunity to test/check if it works.

Unfortunately, it does not seem to.

Here is the screenshot.

It gets stuck with the first file's status saying 'OCR...'.

![Capture](https://user-images.githubusercontent.com/1360723/54065921-5436f380-4238-11e9-804a-5dbb510c7a5e.JPG)

**Comment by niksedk** [comment]
It's probably just slow... ?

**Comment by adem0x** [comment]
The machine I am running SubtitleEdit is AMD ThreadRipper 2 with 128 GB RAM on NVMe storage. IOW, not exactly a slow machine at all. ;)

When I run SubtitleEdit (v3.5.9) [not in batch mode] on that particular file (S01E01.sub), it takes about 7 minutes (with 'Prompt for unknown words' unchecked).

I am guessing, in batch convert mode, SubtitleEdit is trying to display a dialog or some such thing and waiting on user response. And since I don't see any prompt dialogue, it gets stuck.

Then again, this is just a guess. Might be totally wrong/irrelevant.

Anyway, I have about 60 such files. That means, dealing with them in a GUI will be too time consuming.

Is there a command line alternative whereby I can write to convert a given *.sub file to *.srt file?

**Comment by niksedk** [comment]
You could try latest beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.5.9/SubtitleEditBeta.zip

You could also upload the sub (or email it to me), so I can check.

**Comment by adem0x** [comment]
I have downloaded and applied the beta.

It seems to work. But..

Unless one is very VERY patient, it is easy to think 'Subtitle Edit' is stuck because there's hardly any visual feedback that it is actually doing anything --and that goes one for a very long time.

At this point, I suppose it would be useful to add some progress info such as the current file name being processed and which line number it is at.

Adding these shouldn't impact the performance in any significant ways, but would be immensely helpful to the user.

The other thing I have noticed is this: Processing (OCRing) a file in batch mode seems to take more than twice as long than it does compared to standalone 'Subtitle Edit'. This is mostly unexpected --at least to me.

One small feature I would like added is an 'Abort' button to the 'Batch Convert' dialogue. ATM, once I press convert, everything on the dialogue form is disabled. I can minimize it, but not restore it. I have no way other than using task manager to abort the process. In short, an 'Abort' button is sorely needed.

I am not sure what sort of settings 'Subtitle Edit' is using while OCRing .sub files. Does it use any language? If so, which language?

Can it, for example, be made prompt for unknown words and use the user supplied corrections for all the files in the batch?

IOW, could we have another dialog that deals specifically with OCRing?

Finally, since I had about 60 files do convert, I decided to open do them in standalone. I opened ten 'Subtitle Edit's at a time and did the OCRing that way. was a little more labor but it took shorter.

What I did notice while doing that is this: 10 'Subtitle Edit' processes running at the same time didn't put any significant load on my machine --other than each 'Subtitle Edit's display being just a bit sluggish when I moved their windows/forms around. That didn't affect the other processes in any other way at all.

My deduction from that experience is that having ten Tesseract processes running simultaneously on a machine is OK.

That means, I suppose, batch mode OCRing could be multithreaded :)

With the proviso that the user should be given an option to specify the maximum number of threads.
